### PR TITLE
Improving annotation

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,7 +76,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - [#485](https://github.com/nf-core/sarek/pull/485) - `--skip_qc`, `--skip_markduplicates` and `--skip_bqsr` is now `--skip_tools`
 - [#538](https://github.com/nf-core/sarek/pull/538) - `--sequencing_center` is now `--seq_center`
-- [#538](https://github.com/nf-core/sarek/pull/538) - `--markdup_java_options` has been removed
```
Comment (FriederikeHanssen) [inline CHANGELOG.md:79]: did you keep it now afterall?

File: conf/test.config [hunk 2]
```diff
@@ -24,28 +30,25 @@ params {
...
-    intervals               = "${params.genomes_base}/data/genomics/homo_sapiens/genome/genome.interval_list"
-    known_indels            = "${params.genomes_base}/data/genomics/homo_sapiens/genome/vcf/mills_and_1000G.indels.vcf.gz"
     nucleotides_per_second  = 20
 
     snpeff_db         = 'WBcel235.99'
     vep_species       = 'caenorhabditis_elegans'
-    vep_cache_version = '104'
-
-    // Ignore `--input` as otherwise the parameter validation will throw an error
-    schema_ignore_params = 'genomes,input'
```
Comment (FriederikeHanssen) [inline conf/test.config:42]: we don't need this anymore?

File: conf/test.config [hunk 3]
```diff
@@ -24,28 +30,25 @@ params {
 
     // Small reference genome
     igenomes_ignore   = true
-    genome            = 'small_hg38'
-    genomes_base      = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules'
+    genome            = null
+
+    dbsnp             = params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz']
```
Comment (FriederikeHanssen) [inline conf/test.config:35]: 😍

File: CITATIONS.md [hunk 4]
```diff
@@ -71,8 +71,23 @@
   > Danecek P, Auton A, Abecasis G, et al.: The variant call format and VCFtools. Bioinformatics. 2011 Aug 1;27(15):2156-8. doi: 10.1093/bioinformatics/btr330. Epub 2011 Jun 7. PubMed PMID: 21653522; PubMed Central PMCID: PMC3137218.
 
 - [VEP](https://pubmed.ncbi.nlm.nih.gov/27268795/)
+
```
Comment (FriederikeHanssen) [inline CITATIONS.md:74]: Good point need to keep updating this one

File: CHANGELOG.md [hunk 5]
```diff
@@ -76,7 +76,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- [#539](https://github.com/nf-core/sarek/pull/539) - `--annotate_tools` has been removed
+- [#539](https://github.com/nf-core/sarek/pull/539) - `--cadd_cache`, `--cadd_indels`, `--cadd_indels_tbi`, `--cadd_wg_snvs`, `--cadd_wg_snvs_tbi` have been removed
+- [#539](https://github.com/nf-core/sarek/pull/539) - `--genesplicer` has been removed
```
Comment (FriederikeHanssen) [inline CHANGELOG.md:81]: ok so the way to select this now is by setting `--tools vep` and then all the files needed?

File: conf/modules.config [hunk 6]
```diff
@@ -979,56 +987,36 @@ process {
...
+    withName: "NFCORE_SAREK:SAREK:ANNOTATE:.*:TABIX_BGZIPTABIX" {
         publishDir       = [
                 mode: params.publish_dir_mode,
                 path: { "${params.outdir}/annotation/${meta.id}/${meta.variantcaller}" },
                 pattern: "*{gz,gz.tbi}"
         ]
     }
-}
 
-if (params.tools && (params.tools.contains('snpeff') || params.tools.contains('merge'))) {
```
Comment (FriederikeHanssen) [inline conf/modules.config:991]: I guess we still need to find a way for most processes anyways to avoid printing.

File: nextflow_schema.json [hunk 7]
```diff
@@ -49,27 +49,27 @@
...
+                    "help_text": "Multiple separated with commas.\n\nGermline variant calling can currently only be performed with the following variant callers:\n- FreeBayes, HaplotypeCaller, Manta, mpileup, Strelka, TIDDIT\n\nSomatic variant calling can currently only be performed with the following variant callers:\n- ASCAT, Control-FREEC, FreeBayes, Manta, MSIsensorpro, Mutect2, Strelka\n\nTumor-only somatic variant calling can currently only be performed with the following variant callers:\n- Control-FREEC, Manta, mpileup, Mutect2, TIDDIT\n\nAnnotation is done using snpEff, VEP, or even both consecutively.\n\n> **NB** As Sarek will use bgzip and tabix to compress and index VCF files annotated, it expects VCF files to be sorted.\n\n\n\n`DNAseq`, `DNAscope` and `TNscope` are only available with `--sentieon`\n\n> **NB** tools can be specified with no concern for case.",
```
Comment (FriederikeHanssen) [inline nextflow_schema.json:59]: VEP and snpeff results are merged. Should that be added to the text? Or is that what you mean with consecutively? I would understand it as `--tools snpeff,VEP` 

Snetieon can be removed for now, right? It is not going to be part of the release

**Review by FriederikeHanssen — APPROVED** [review]
Love it, just some small question so I understand the implementation better

**Comment by FriederikeHanssen** [comment]
also not sure why variant_calling test and strelka is failing, this shouldn't be the case

File: CHANGELOG.md (hunk 1 above)
Comment (maxulysse) [reply CHANGELOG.md:79]: no, I must have commited the wrong version of `CHANGELOG`

File: conf/modules.config (hunk 6 above)
Comment (maxulysse) [reply conf/modules.config:991]: This will be solve by Nextflow at some point

File: nextflow_schema.json (hunk 7 above)
Comment (maxulysse) [reply nextflow_schema.json:59]: That's what I mean by consecutively.
I actually mean that the snpeff annotated vcf gets annotated by VEP, but was never really sure on how to phrase that.

Good point about sentieon, I'll do that here as well

File: CHANGELOG.md (hunk 5 above)
Comment (maxulysse) [reply CHANGELOG.md:81]: yeah, will need some more docs about it, but all files that we supply for VEP go into this vep_extra_files channel.
Need to check out if it behave properly and don't get consume whenever

File: conf/test.config (hunk 2 above)
Comment (maxulysse) [reply conf/test.config:42]: Will verify that once more, now that I see it, I'm wondering the same

File: conf/modules.config (hunk 6 above)
Comment (maxulysse) [reply conf/modules.config:991]: Or should anyway

File: conf/test.config (hunk 3 above)
Comment (maxulysse) [reply conf/test.config:35]: don't know why we didn't do this before

File: nextflow_schema.json (hunk 7 above)
Comment (maxulysse) [reply nextflow_schema.json:59]: better do it in a separate PR

**Review by FriederikeHanssen — APPROVED** [review]

**Comment by maxulysse** [comment]
@nf-core-bot fix linting pretty please

**Pull request merged**